allennlp/state_machines/interactive_beam_search.py

```diff
@@ -0,0 +1,143 @@
+from collections import defaultdict
+from typing import Dict, List, Tuple
+
+import torch
+
+from allennlp.state_machines import util
+from allennlp.state_machines.beam_search import BeamSearch, StateType
+from allennlp.state_machines.transition_functions import TransitionFunction
+
+
+class InteractiveBeamSearch(BeamSearch[StateType]):
```

Comment:
So, now it looks like this code is basically identical to `BeamSearch`, except you have an optional constrained prefix.  This is like a 10 line addition to `BeamSearch`, and it would be generally useful in some cases (e.g., if you know the answer type at training time, you can constrain the first decision).  What do you think of just adding this to that code, and removing this additional class altogether?

The one thing I'm not sure of is your `self.beams` field, which looks like it keeps around the history at every timestep, instead of at the end.  We could add that, too, with something like a `keep_beam_history` parameter.

Reply:
done